okapi-core/src/main/java/org/folio/okapi/managers/InternalModule.java

```diff
@@ -750,22 +751,22 @@ private String location(ProxyContext pc, String id, String baseUri, String s) {
   private Future<String> createTenant(ProxyContext pc, String body) {
     try {
       final TenantDescriptor td = JsonDecoder.decode(body, TenantDescriptor.class);
-      if (td.getId() == null || td.getId().isEmpty()) {
-        td.setId(UUID.randomUUID().toString());
-      }
-      final String tenantId = td.getId();
-      if (!tenantId.matches("^[a-z0-9_-]+$")) {
-        return Future.failedFuture(
-            new OkapiError(ErrorType.USER, messages.getMessage("11601", tenantId)));
-      }
-      Tenant t = new Tenant(td);
-      return tenantManager.insert(t).map(res ->
-        location(pc, tenantId, null, Json.encodePrettily(t.getDescriptor())));
+      return validateTenantId(td)
+          .compose(x -> tenantManager.insert(new Tenant(td)))
+          .map(x -> location(pc, td.getId(), null, Json.encodePrettily(td)));
     } catch (DecodeException ex) {
       return Future.failedFuture(new OkapiError(ErrorType.USER, ex.getMessage()));
     }
   }
 
+  static Future<Void> validateTenantId(TenantDescriptor td) {
+    if (td.getId() == null || td.getId().isEmpty()) {
+      td.setId(("t" + UUID.randomUUID().toString().replace("-", ""))
```

Comment:
Maybe break creation of the uuid tenant id out into its own method called `tryCreateTenantId` or simlar and do validation separately since they seem like different operations to me.